vendor/gx/ipfs/QmTRhk7cgjUf2gfQ3p2M9KPECNZEW9XUrmHcFCgog4cPgB/go-libp2p-peer/peer.go

```diff
@@ -62,12 +62,17 @@ func (id ID) MatchesPrivateKey(sk ic.PrivKey) bool {
 }
 
 // MatchesPublicKey tests whether this ID was derived from pk
+// OpenBazaar: test both inline and and hashes pubkeys.
 func (id ID) MatchesPublicKey(pk ic.PubKey) bool {
 	oid, err := IDFromPublicKey(pk)
 	if err != nil {
 		return false
 	}
-	return oid == id
+	iid, err := InlineIDFromPublicKey(pk)
+	if err != nil {
+		return false
+	}
+	return oid == id || iid == id
```

Comment:
That is a good point but I think the only thing that would cause an error in `IDFromPublicKey` is the `Bytes()` method erroring but I don't think that should ever happen. If it does both functions will error since they both read from bytes.